customcontext/context.h

```diff
@@ -59,34 +59,53 @@
 #include "threaduploadtexture.h"
 #endif
 
+#if QT_VERSION >= QT_VERSION_CHECK(5, 8, 0)
+#include <private/qsgdefaultcontext_p.h>
+#include <private/qsgdefaultrendercontext_p.h>
+#endif
 
-#if QT_VERSION >= 0x050200
 struct QSGMaterialType;
+
+#if QT_VERSION >= QT_VERSION_CHECK(5, 8, 0)
+#define CONTEXT_CLASS_BASE QSGDefaultContext
+#define RENDER_CONTEXT_CLASS_BASE QSGDefaultRenderContext
+#define COMPILESHADER_METHOD compileShader
+#else
+#define CONTEXT_CLASS_BASE QSGContext
+#define RENDER_CONTEXT_CLASS_BASE QSGRenderContext
+#define COMPILESHADER_METHOD compile
 #endif
 
 namespace CustomContext
 {
 
-#if QT_VERSION >= 0x050200
-class RenderContext : public QSGRenderContext
+class RenderContext : public RENDER_CONTEXT_CLASS_BASE
 {
 public:
     RenderContext(QSGContext *ctx);
-    void initialize(QOpenGLContext *context);
-    void invalidate();
+
+#if QT_VERSION >= QT_VERSION_CHECK(5, 8, 0)
+    void renderNextFrame(QSGRenderer *renderer, uint fbo);
+#else
     void renderNextFrame(QSGRenderer *renderer, GLuint fbo);
+#endif
 
-#if QT_VERSION < 0x050600
-    QSGTexture *createTexture(const QImage &image) const;
-    QSGTexture *createTextureNoAtlas(const QImage &image) const;
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
+    void initialize(const InitParams *inCtx);
+#elif (QT_VERSION >= QT_VERSION_CHECK(5, 8, 0) && QT_VERSION < QT_VERSION_CHECK(5, 14, 0))
```

Comment:
The #if above should already guarantee the version is below 5.14, and skipping that would make the rules similar between cpp and this.

Reply:
OK removed it again